.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,26 @@
+name: CI Linter with golangci-lint
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: Set up golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.20.6
+
+    - name: Run Linter
+      run: make golangci-lint
```

Comment:
let's start like this, but I'm eyeing the official action for lesser resource usage, saving us CI time.